File: Switch/app/src/main/java/com/example/aswitch/MainActivity.java
package com.example.aswitch;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    EditText edtNumber;
    Button btnSub;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initview();

    }

    private void initview()
    {
        edtNumber=findViewById(R.id.edtNumber);
        btnSub=findViewById(R.id.btnSub);

        btnSub.setOnClickListener(v ->{
            String v1=edtNumber.getText().toString();

            Toast.makeText(this,""+v1,Toast.LENGTH_SHORT).show();
        });
    }
}

File: CompanionObjects/app/src/main/java/com/example/companionobjects/MainActivity.kt
package com.example.companionobjects

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.text.method.PasswordTransformationMethod
import android.widget.Button
import android.widget.EditText
import com.example.companionobjects.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var editTextPassword: EditText
    private lateinit var toggleButtonShowPassword: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initView()
    }

    private fun initView() {
        editTextPassword = findViewById(R.id.editTextPassword)
        toggleButtonShowPassword = findViewById(R.id.toggleButtonShowPassword)

        toggleButtonShowPassword.setOnClickListener {
            togglePasswordVisibility()
        }
    }

    private fun togglePasswordVisibility() {
        if (editTextPassword.transformationMethod is PasswordTransformationMethod) {
            // Password characters are hidden, show them
            editTextPassword.transformationMethod = null
        } else {
            // Password characters are shown, hide them
            editTextPassword.transformationMethod = PasswordTransformationMethod()
        }

        // Move the cursor to the end of the input to maintain the user's cursor position
        editTextPassword.setSelection(editTextPassword.text.length)
    }
}
